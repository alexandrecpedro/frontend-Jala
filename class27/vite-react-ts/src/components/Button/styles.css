.btn {
  background-color: #34c703;
  color: azure;
  font-size: 1.25rem;
  width: 18.9rem;
  height: 2.8rem;
  display: flex;
  justify-content: center;
  align-items: center;
  border: none;
  border-radius: 5px;
  transition: all 0.2s ease;

  &:hover {
    background-color: #185a02;
  }
}
