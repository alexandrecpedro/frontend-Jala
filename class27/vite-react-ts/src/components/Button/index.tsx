import React from 'react';
import './styles.css';

interface ButtonProps {
  readonly text: string;
}

const Button: React.FC<ButtonProps> = ({ text }) => {
  return (
    //for accessibility purposes we have type and aria-label well defined here.
    <button className='btn' type="submit" aria-label={text}>
      {text}
    </button>
  );
};

export default Button;
