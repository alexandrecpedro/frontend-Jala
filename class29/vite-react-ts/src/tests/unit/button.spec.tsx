// button.spec.tsx
import renderer from 'react-test-renderer'
import Button from '../../components/Button';

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  console.warn(result)
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

test('Button component', () => {
    const component = renderer.create(<Button onClick={() => {}} text="Click me" />);   
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
    expect(tree.type).toBe('button')
});

test('Button component', () => {
    const component = renderer.create(<Button onClick={() => {}} text="Click me" />);   
    const tree = toJson(component)
    tree.props.onClick()
    expect(tree.props['aria-label']).toBe('Click me')
});

