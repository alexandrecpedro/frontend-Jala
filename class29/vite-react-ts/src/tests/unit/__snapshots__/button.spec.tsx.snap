// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component 1`] = `
<button
  aria-label="Click me"
  onClick={[Function]}
  type="button"
>
  Click me
</button>
`;
